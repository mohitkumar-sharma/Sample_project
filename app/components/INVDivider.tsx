import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native';
import config from '../config';

export interface INVDividerProps {
  /* as an object of style props for View element */
  dividerStyle?: StyleProp<ViewStyle>;
}

/**
 * This is the INVDivider functional component works as a divider and can be used within the app
 * @param props : props are defined in INVDividerProps which can be passed to this component
 */
const INVDivider = (props: INVDividerProps): JSX.Element => {
  const { dividerStyle } = props;
  return <View style={[styles.divider, dividerStyle]} />;
};

const styles = StyleSheet.create({
  divider: { height: 1, backgroundColor: config.colors.divider },
});

export default INVDivider;
