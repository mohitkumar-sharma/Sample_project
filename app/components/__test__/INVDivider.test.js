import React from 'react';
import { render } from '@testing-library/react-native';
import INVDivider from '../INVDivider';

describe('INVDivider', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(<INVDivider />);
    expect(toJSON()).toMatchSnapshot();
  });
});
