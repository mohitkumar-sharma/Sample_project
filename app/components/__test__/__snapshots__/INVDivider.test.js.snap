// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INVDivider matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;
